crates/bitcoin/src/parser.rs

```diff
@@ -224,21 +223,24 @@ pub trait FromLeBytes: Sized {
 
 impl FromLeBytes for BlockHeader {
     fn from_le_bytes(bytes: &[u8]) -> Result<BlockHeader, Error> {
-        parse_block_header(&RawBlockHeader::from_bytes(bytes)?)
+        parse_block_header(bytes)
     }
 }
 
-// like parse_block_header, but without the version check. This is needed in testing, to test
-// with historical data
-pub fn parse_block_header_lenient(raw_header: &RawBlockHeader) -> Result<BlockHeader, Error> {
-    let mut parser = BytesParser::new(raw_header.as_bytes());
+/// Parses the raw bitcoin header into a Rust struct
+///
+/// # Arguments
+///
+/// * `header` - An 80-byte Bitcoin header
+pub fn parse_block_header(bytes: &[u8]) -> Result<BlockHeader, Error> {
```

Comment:
I think we may need to reuse some of this code downstream, so I would be inclined to leave that for now and revisit once I know for sure it will not be required.

Reply:
> I think we may need to reuse some of this code downstream, so I would be inclined to leave that for now and revisit once I know for sure it will not be required.

Downstream can enable the feature?